SHS-6205: Dashboard | "My latest edits" should show date of my edit

nJim opened the pull request:
# Summary
- Create a new view to show all node revisions and replaces the existing 'My latest edits' block with this new one on the dashboard.

## Backstory
- This is part of an initiative to refine the recently launch dashboard work.
- Currently the 'My latest edits' block is not behaving in a helpful way. It is only showing information about latest revisions. Because of this, when a node is updated by another user, the revision falls off this display.
- A new view is created based on revisions table (instead of the node table) so that this block can display a longer running list of all revisions associated with this user.

## Need Review By (Date)
- Not set.

## Urgency
- Not set.

## Steps to Test
1. [Log into a tugboat environment as an admin user.](https://pr1876-zs2lwyw8pizkmbnwpspqz4z6iefzh53f.tugboatqa.com/)
2. [Edit a flexible page](https://pr1876-zs2lwyw8pizkmbnwpspqz4z6iefzh53f.tugboatqa.com/node/17891/edit). And save the changes.
3. [Visit the dashboard](https://pr1876-zs2lwyw8pizkmbnwpspqz4z6iefzh53f.tugboatqa.com/admin/dashboard) and locate the 'My Latest Edits' block. 
  i. Verify that the recently updated node appears at the top of this list
  ii. The title links to the latest revision
  iii. The operation links are for the latest version of the node
4. Test multiple changes: Edit the page again and save the node. Verify that it only appears once in the list of recent changes.
5. Test title change: Change the node title. Verify that it only appears once in the block.
6. Test multiple authors: Test editing one of these nodes as another author. Check that the revisions do not drop off your dashboard when someone else makes an edit. 
  i. Also, this list is sorted by the data of your last revision. If someone else updates a node on this list, your sort order is not changed.

# Review Tasks

## Backend / Functional Validation
### Code
- [ ] Are the naming conventions following our standards?
- [ ] Are PHP functions and variables in `snake_case` and not `camelCase`?
- [ ] Does Drupal code follow [Drupal Coding Standards](https://www.drupal.org/docs/develop/standards/php/php-coding-standards)?
- [ ] Does the code have sufficient inline comments?
- [ ] Is there anything in this code that would be hidden or hard to discover through the UI?
- [ ] Are there any [code smells](https://blog.codinghorror.com/code-smells/)?
- [ ] Are tests provided?

### Code security
- [ ] Is all [user input properly sanitized when rendered](https://www.drupal.org/docs/8/security/drupal-8-sanitizing-output)?
- [ ] Any obvious [security flaws or new areas for attack](https://www.drupal.org/docs/8/security)?

### General
- [ ] Is there anything included in this PR that is not related to the problem it is trying to solve?
- [ ] Is the approach to the problem appropriate?


---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1210264640232270

nJim committed:
fix(SHS-6205): update latest commits to show all revisions

ahughes3 commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1876-dtnu3klqnrotyfuaxzxb3mvtmfj3lxaq.tugboatqa.com
- https://hs-colorful-dtnu3klqnrotyfuaxzxb3mvtmfj3lxaq.tugboatqa.com
- https://hs-traditional-dtnu3klqnrotyfuaxzxb3mvtmfj3lxaq.tugboatqa.com
- https://history-dtnu3klqnrotyfuaxzxb3mvtmfj3lxaq.tugboatqa.com
- https://west-dtnu3klqnrotyfuaxzxb3mvtmfj3lxaq.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/68562d1581f706d992ca4d88

Visual Diffs:
- https://dashboard.tugboatqa.com/68562d1581f706d992ca4d88/visualdiffs

nJim committed:
feat(SHS-6205): reduce duplicate revisions

cienvaras reviewed (changes requested):
@nJim The block seems to work as expected, but found a couple of things that need to be checked:
- I cannot see the block as a "Site Manager" user (tested masquerading as "Lindsey" in the test site). All users with access to the dashboard ("Developer", "Site Manager" and "Contributor" roles) should be able to see and interact with this block.
- The block should have the same "no results" behavior that the previous block had.
- The old view (`my_latest_edits`) can be removed, no need to keep it.

nJim committed:
feat(SHS-6205): remove previous view

ahughes3 commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1876-dtnu3klqnrotyfuaxzxb3mvtmfj3lxaq.tugboatqa.com
- https://hs-colorful-dtnu3klqnrotyfuaxzxb3mvtmfj3lxaq.tugboatqa.com
- https://hs-traditional-dtnu3klqnrotyfuaxzxb3mvtmfj3lxaq.tugboatqa.com
- https://history-dtnu3klqnrotyfuaxzxb3mvtmfj3lxaq.tugboatqa.com
- https://west-dtnu3klqnrotyfuaxzxb3mvtmfj3lxaq.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/68562d1581f706d992ca4d88

Visual Diffs:
- https://dashboard.tugboatqa.com/68562d1581f706d992ca4d88/visualdiffs

nJim committed:
Merge branch '11.12.1-release' into SHS-6205-latest-edits

ahughes3 commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1876-qc8qagkc9dtaqjqmgguzv7jjhsxpevc5.tugboatqa.com
- https://hs-colorful-qc8qagkc9dtaqjqmgguzv7jjhsxpevc5.tugboatqa.com
- https://hs-traditional-qc8qagkc9dtaqjqmgguzv7jjhsxpevc5.tugboatqa.com
- https://history-qc8qagkc9dtaqjqmgguzv7jjhsxpevc5.tugboatqa.com
- https://west-qc8qagkc9dtaqjqmgguzv7jjhsxpevc5.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/68635eb5a8c62ea4914502cd

Visual Diffs:
- https://dashboard.tugboatqa.com/68635eb5a8c62ea4914502cd/visualdiffs

nJim committed:
feat(SHS-6205): style more link

ahughes3 commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1876-9it6zo47ck82bhjzesmr1fbte2kc2nqv.tugboatqa.com
- https://hs-colorful-9it6zo47ck82bhjzesmr1fbte2kc2nqv.tugboatqa.com
- https://hs-traditional-9it6zo47ck82bhjzesmr1fbte2kc2nqv.tugboatqa.com
- https://history-9it6zo47ck82bhjzesmr1fbte2kc2nqv.tugboatqa.com
- https://west-9it6zo47ck82bhjzesmr1fbte2kc2nqv.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/68640f99c18ac92245590b34

nJim commented:
@cienvaras This is ready for review again.
> * I cannot see the block as a "Site Manager" user (tested masquerading as "Lindsey" in the test site). All users with access to the dashboard ("Developer", "Site Manager" and "Contributor" roles) should be able to see and interact with this block.
I've updated this new view to match the same permissions as the previous block.

> * The block should have the same "no results" behavior that the previous block had.
I added the no-results message as well as a view-more link so that this new view matches the previous one.

> * The old view (`my_latest_edits`) can be removed, no need to keep it.
Done.

cienvaras committed:
Merge branch '11.12.1-release' of github.com:SU-HSDO/suhumsci into SHS-6205-latest-edits

cienvaras committed:
chore(shs-6205): add update hook to remove my_latest_edits unused view

cienvaras reviewed (approved):
@nJim Looks good, thanks for the fixes! Just one thing for future tickets: this platform's CI runs config import with the `--partial` flag, meaning that any deleted config won't be removed from the DB. If we want to completely remove something, it needs to be done in an update hook. In this case, it was a really simple one, so I quickly added it while testing.

@ahughes3 Ready for you.

Jnwibowo7321 reviewed (approved):
LGTM!

ahughes3 commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1876-9it6zo47ck82bhjzesmr1fbte2kc2nqv.tugboatqa.com
- https://hs-colorful-9it6zo47ck82bhjzesmr1fbte2kc2nqv.tugboatqa.com
- https://hs-traditional-9it6zo47ck82bhjzesmr1fbte2kc2nqv.tugboatqa.com
- https://history-9it6zo47ck82bhjzesmr1fbte2kc2nqv.tugboatqa.com
- https://west-9it6zo47ck82bhjzesmr1fbte2kc2nqv.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/68640f99c18ac92245590b34

cienvaras committed:
Merge branch '11.12.2-release' of github.com:SU-HSDO/suhumsci into SHS-6205-latest-edits

ahughes3 commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1876-9it6zo47ck82bhjzesmr1fbte2kc2nqv.tugboatqa.com
- https://hs-colorful-9it6zo47ck82bhjzesmr1fbte2kc2nqv.tugboatqa.com
- https://hs-traditional-9it6zo47ck82bhjzesmr1fbte2kc2nqv.tugboatqa.com
- https://history-9it6zo47ck82bhjzesmr1fbte2kc2nqv.tugboatqa.com
- https://west-9it6zo47ck82bhjzesmr1fbte2kc2nqv.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/68640f99c18ac92245590b34
